src/data/nat/gcd.lean

```diff
@@ -136,6 +136,9 @@ by rw [gcd_comm, gcd_gcd_self_right_right]
 @[simp] lemma gcd_gcd_self_left_left (m n : ℕ) : gcd (gcd m n) m = gcd m n :=
 by rw [gcd_comm m n, gcd_gcd_self_left_right]
 
+lemma gcd_add_self (m n k : ℕ) : gcd m n = gcd m (n + k * m) :=
```

Comment:
I suspect this (as well as `gcd_fib_fib`, `gcd_fib_fib'`, `gcd_fib_add_self`, `fib_add`) should be the other way round to make `rw` with them easier